### PR TITLE
[User's personal cabinet] Authority doesn't set

### LINKED ISSUE #6135: [User's personal cabinet]An employee in the position of Service Manager does not have access to the function Create/Edit/Deactivate/an employee card, Edit an employee`s authorities
Environment: Windows 10 Pro 20H2
Google Chrome 113.0.5672.93
Reproducible: always
Build found: 31.07.2023
Preconditions

1. Super Admin should add an employee with a Service Manager position
2. The checkboxes "Set Employee permissions" should be checked by default
3. Super Admin in "Set Employee permissions" checks on the checkboxes "Create a new employee card", "Edit an employee card", "Deactivate an employee", "Edit an employee`s authorities"
4. The user should be logged in https://www.greencity.social/#/greenCity as a Service Manager

Steps to reproduce

1. Click on the icon in the upper right corner
2. In pop-up window click on "UBS-ADMIN"
3. Click on the "Employees tab"

Actual result

1. Service Manager does not have access to the function "Create a new employee card"
2. Service Manager does not have access to the function "Edit an employee card"
3. Service Manager does not have access to the function "Deactivate an employee"
4. Service Manager does not have access to the function "Edit an employee`s authorities"

Expected result

1. Service Manager has access to the function "Create a new employee card"
2. Service Manager has access to the function "Edit an employee card"
3. Service Manager has access to the function "Deactivate an employee"
4. Service Manager has access to the function "Edit an employee`s authorities"

User story https://github.com/ita-social-projects/GreenCity/issues/4617

"Bug", Priority ("Medium"), Severity ("Major"), Type ("functional"), "API" (for back-end bugs)
Create Edit Deactivate employee card
Not access to create an employee card

### PR DESCRIPTION
**Pull request by ospodaryk**
# GreenCityUser PR
[[User's personal cabinet] Authority doesn't set](https://github.com/ita-social-projects/GreenCity/issues/6135)

## Summary Of Changes :fire:
Added to position new field nameEn.

## Added
Added to position new field nameEn.
Added liquibase to change structure of Position table.

## Changed
Position Dto and method in PositionRepo
Changed tests to test User Repo.

# PR Checklist Forms

_(to be filled out by PR submitter)_
- [ ] Code is up-to-date with the `dev` branch.
- [x] You've successfully built and run the tests locally.
- [x] There are new or updated unit tests validating the change.
- [x] JIRA/ Github Issue number & title in PR title (ISSUE-XXXX: Ticket title)
- [x] This template filled (above this section).
- [x] Sonar's report does not contain bugs, vulnerabilities, security issues, code smells ar duplication
- [x] `NEED_REVIEW` and `READY_FOR_REVIEW` labels are added.
- [x] All files reviewed before sending to reviewers

### CODE REVIEW

File: dao/pom.xml [hunk 1]
```diff
@@ -16,6 +16,18 @@
 
     <artifactId>dao</artifactId>
     <dependencies>
+        <!-- liquibase -->
```
Comment (RomaMocherniuk) [inline dao/pom.xml]: this comment is redundant

Comment (ospodaryk) [reply dao/pom.xml]: removed

**Review by ABbondar — APPROVED** [review]

File: service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java [hunk 2]
```diff
@@ -220,6 +220,10 @@ public SuccessSignUpDto signUpEmployee(EmployeeSignUpDto employeeSignUpDto, Stri
         employee.setShowShoppingList(true);
         List<String> positionNames = employeeSignUpDto.getPositions().stream()
             .map(PositionDto::getName).collect(Collectors.toList());
+        if (positionNames.isEmpty()) {
```
Comment (RomaMocherniuk) [inline service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java]: Why don't we use both `name` and `naneEn` for `positionsNames`?

File: service/src/test/java/greencity/repository/UserRepoTest.java [hunk 3]
```diff
@@ -0,0 +1,310 @@
...
+
+@ExtendWith(MockitoExtension.class)
+class UserRepoTest {
+    @Mock
+    UserRepo userRepo;
+
+    @Test
+    void findByEmailTest() {
+        Long expected = 1L;
+        when(userRepo.findByEmail("test@email.com")).thenReturn(Optional.of(ModelUtils.getUser()));
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/repository/UserRepoTest.java:44]: please add `verify`

File: service/src/test/java/greencity/repository/UserRepoTest.java [hunk 4]
```diff
@@ -0,0 +1,310 @@
...
+    }
+
+    @Test
+    void findAllTest() {
+        Pageable pageable = PageRequest.of(0, 3);
+        List<User> mockUsers = Arrays.asList(ModelUtils.getUser(), ModelUtils.getUser(), ModelUtils.getUser());
+
+        Page<User> expectedPage = new PageImpl<>(mockUsers, pageable, mockUsers.size());
+
+        when(userRepo.findAll(any(Pageable.class))).thenReturn(expectedPage);
```
Comment (RomaMocherniuk) [inline service/src/test/java/greencity/repository/UserRepoTest.java:57]: please add verify for all tests

File: service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java (hunk 2 above)
Comment (ospodaryk) [reply service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java]: changed method

File: service/src/test/java/greencity/repository/UserRepoTest.java (hunk 3 above)
Comment (ospodaryk) [reply service/src/test/java/greencity/repository/UserRepoTest.java:44]: added

File: service/src/test/java/greencity/repository/UserRepoTest.java (hunk 4 above)
Comment (ospodaryk) [reply service/src/test/java/greencity/repository/UserRepoTest.java:57]: added

File: service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java (hunk 2 above)
Comment (ospodaryk) [reply service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java]: changed method

**Review by 2or5 — APPROVED** [review]

**Review by RomaMocherniuk — APPROVED** [review]

**Review by Yurii-Midianyi — APPROVED** [review]

**Pull request merged**